src/Maestro/Client/src/MaestroApiCredential.cs

```diff
@@ -87,6 +87,24 @@ internal static MaestroApiCredential CreateFederatedCredential(string barApiBase
             return new MaestroApiCredential(credential, requestContext);
         }
 
+        /// <summary>
+        /// Use this for darc invocations from services using an MI
+        /// </summary>
+        internal static MaestroApiCredential CreateManagedIdentityCredential(string barApiBaseUri, string managedIdentityId)
+        {
+            string appId = EntraAppIds[barApiBaseUri.TrimEnd('/')];
+
+            var miCredential = new ManagedIdentityCredential(managedIdentityId);
```

Comment:
```suggestion
            ManagedIdentityCredential miCredential = new(managedIdentityId);
```
aren't we trying to use the new syntax everywhere?